src/error_chain.rs

```diff
@@ -137,6 +137,17 @@ macro_rules! error_chain_processed {
                 )
             }
 
+            /// Construct a chained error from another boxed error and a kind, and generates a backtrace
+            pub fn with_boxed_chain<K>(error: Box<::std::error::Error + Send>, kind: K)
+                -> $error_name
+                where K: Into<$error_kind_name>
+            {
+                $error_name(
+                    kind.into(),
+                    $crate::State::new::<$error_name>(error, ),
+                )
+            }
+
```

Comment:
Ok, I rewrote `with_chain` to use `with_boxed_chain`, and I updated the changelog! I will be monitoring the situation with `with_boxed_chain`, specialization, etc, since I really want to improve the API for working with boxed trait errors!

I've tested these changes against another major internal codebase, and it all seems to work.

I think this is ready to merge. Let me know if I should add anything else; otherwise, hopefully you'll see a new PR in a few months from me handling the Error specialization correctly :)